src/main/java/org/folio/circulation/resources/OverrideCheckOutStrategy.java

```diff
@@ -0,0 +1,85 @@
+package org.folio.circulation.resources;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.circulation.domain.representations.CheckOutByBarcodeRequest.ITEM_BARCODE;
+import static org.folio.circulation.support.Result.failed;
+import static org.folio.circulation.support.Result.succeeded;
+import static org.folio.circulation.support.ValidationErrorFailure.singleValidationError;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.LoanAndRelatedRecords;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.http.server.ValidationError;
+import org.joda.time.DateTime;
+
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Checkout strategy for non loanable items.
+ * Sets due date passed in request body.
+ * Refuses request if an item is loanable
+ */
+public class OverrideCheckOutStrategy implements CheckOutStrategy {
+
+  private static final String DUE_DATE = "dueDate";
+  private static final String COMMENT = "comment";
+
+
+  @Override
+  public CompletableFuture<Result<LoanAndRelatedRecords>> checkOut(LoanAndRelatedRecords relatedRecords,
+                                                                   JsonObject request,
+                                                                   Clients clients) {
+    String comment = request.getString(COMMENT);
```

Comment:
I think this is a significant improvement over the inheritance model. Thank you for making this change.

This does feel like a little bit of a clunky abstraction, in that it is both processing the check out _and_ validating the incoming request.

Splitting it into two methods might help.

I think more substantial changes may need to be deferred. I think there are other parts of the check out process that could move to these classes, as an overridden check out likely does not need some of the steps of a regular check out.

Does that make sense?

Reply:
Makes sense to me. I'm not familiar enough with the logic in checkout resource so I'm not sure how to better split it.  Would you like this changes to be made it this pull request, or left this for further refactoring?

Reply:
I think it may as well wait till another time